components/datetime/src/format/time_zone.rs

```diff
@@ -54,258 +54,6 @@ where
     T: TimeZoneInput,
     W: fmt::Write + ?Sized,
 {
-    match &time_zone_format.kind {
```

Comment:
I believe the whole removed code can be replaced by `format_with_fallback`, which is still in progress.